### DISCUSSION TITLE
Errors in rosetta_scripts.static.linuxgccrelease-missing heavyatom

### DISCUSSION THREAD

**Discussion by linbaixin** [RosettaCommons/rosetta#133] [Developing]
[output.txt](https://github.com/user-attachments/files/16762301/output.txt)
After I submitted the RosettaScript program, the output file indicated that some atoms were missing from the ligand, but the generated parameter file shows that its structure is complete."

**Comment by roccomoretti** [comment]
The message you get (`core.conformation.Conformation: [ WARNING ] missing heavyatom:` and `ERROR: too many tries in fill_missing_atoms!` is due to there not being enough atoms in the input PDB, not (necessarily) due to any issue with the parameter file.

In your particular case, though, I note that you're getting a message `core.conformation.Conformation: [ WARNING ] missing heavyatom: N1 on residue pdb_LIG 333`  -- the `pdb_` prefix on the residue type name indicates that you're attempting to use the wwPDB's Chemical Component Dictionary definition of the `LIG` residue type, rather than any custom defined residue. -- The reason for this is your command line. If you have a params file generated for the LIG residue, you need to provide it on the command line with the `-extra_res_fa` option to tell Rosetta to use it.

I'll note that you also have atom naming issues with residue 332 -- Rosetta is thinking your 'SAM' residue is S-adenosyl_methionine, but the naming conventions it expects is not the naming convention you're providing. You either need to match the naming convention, or you need to provide a custom parameter file for the SAM residue as well.
